clientd/src/lib.rs

```diff
@@ -114,15 +114,144 @@ pub async fn run_dispatcher(
 
 async fn handle_rpc_request(
     request: RpcRequest,
-    fedimint_client: &Client<UserClientConfig>,
+    fedimint_client: Option<&FedimintClient>,
 ) -> Result<HandlerResult, CallError> {
     match request {
         RpcRequest::HealthCheck() => Ok(HandlerResult::HealthCheck()),
-        RpcRequest::Info() => rpc::info(fedimint_client).await,
+        RpcRequest::Info() => {
+            rpc::info(
+                fedimint_client
+                    .expect("fedimint client was passed to the dispatcher")
+                    .as_ref(),
```

Comment:
Why make the fn arg optional? It just add failure cases.

Reply:
Testing. I could only do "real" integration tests and no unit tests if it was required. I could also mock the client and pass a `dyn FederationClient` but that would be weird.